cou/apps/auxiliary.py

```diff
@@ -457,3 +465,142 @@ async def _verify_nova_compute(self, target: OpenStackRelease) -> None:
             raise ApplicationError(
                 f"Units '{', '.join(units_not_upgraded)}' did not reach {target}."
             )
+
+
+@AppFactory.register_application(["vault"])
+class Vault(AuxiliaryApplication):
+    """Application for vault."""
+
+    wait_timeout = LONG_IDLE_TIMEOUT
+    wait_for_model = True
+
+    def _check_snap_vault_installed(self) -> None:
+        """Make sure snap vault is installed.
+
+        :raises SnapVaultNotInstalled: When no snap valut installed.
+        """
+        result = subprocess.run(
+            "snap info vault".split(),
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            text=True,
+            check=True,
+        )
+        if "installed" not in result.stdout:
+            raise SnapVaultNotInstalled("snap vault is required to upgrade the vault")
+
+    async def _wait_for_sealed_status(self) -> None:
```

Comment:
since you created `steps/vault.py` I would recommend that you move vault related function to that file

Reply:
What we have in `step/vault.py` is a pre-check before all the upgrading.

The required step will be in the `Vault` o7k application class.